### ISSUE TITLE
Feat: Publish TLMs asynchronously

### ISSUE THREAD

**Issue by grignards** [tracim/tracim#3100]
## Feature description

The TLMs must be published asynchronously when Tracim's configuration is set to `async`:

- [x] Rename the following configuration parameters' names to `jobs.xxx`:
```ini
; email.processing_mode = sync
## with async, please also configure redis below
; email.async.redis.host = localhost
; email.async.redis.port = 6379
; email.async.redis.db = 0
```
- [x] use an RQ job when `jobs.processing_mode` is `async`. This requires the custom RQ worker realized in #2991
- [x] directly publish the messages when `jobs.processing_mode` is `sync` as currently. The current "quick'n dirty" code for this is functional but not very clean (see #3064 for instance).
- [x] update documentation about the configuration parameters (`doc/live_message_setup.md`, `doc/setting.md`)
- [x] make code thread-safe : https://github.com/tracim/tracim/issues/3064
## Context

The creation/publication of TLMs can take time so deferring it to a background job is a useful option.

## Prerequisites

- #2991 must be merged

## Impacted components

- *backend*

## Extra information

Message publishing code is in `lib.core.event.py`, `EventBuilder._publish_event`.

**Comment by inkhey** [comment]
1.As this do required to change some of the email config, it make sense to check also this issue about refactoring the email config logic: https://github.com/tracim/tracim/issues/832 . Depending on how deep we change the logic, it make and does not make sense to solve it at the same time.

2.Also, as we do get probably multiple async worker (see https://github.com/tracim/tracim/issues/2901) does it make sense to rename some email.* to live_messages.* as some worker may be not be directly live_messages related ? Should we get a config "per worker" or a global config for all redis based workers ? I'm not sure we do need to allow very complicated setup with some worker in sync, some other in sync, etc…

**Comment by grignards** [comment]
> 1. As this does required to change some of the email config, it make sense to check also this issue about refactoring the email config logic: #832 . Depending on how deep we change the logic, it make and does not make sense to solve it at the same time.

In this issue the email changes are "only" renaming of the config parameters. It should be done before #2901 where most of the email related work will be done.

> 2. Also, as we do get probably multiple async worker (see #2901) does it make sense to rename some email.* to live_messages.* as some worker may be not be directly live_messages related ? Should we get a config "per worker" or a global config for all redis based workers ? I'm not sure we do need to allow very complicated setup with some worker in sync, some other in async, etc…

I think we should not over-complicate things here: either we are async or not, having a setting by job class is overengineering IMHO as the main point of having the sync mode is to have an easier deployment option.

> 3. Related to previous discussion with @grignards, I do suggest the idea, as pushpin will be now mandatory, that worker with redis will be also mandatory. The main drawback i see with this idea is that make things more complicated for test.

Yes that's why I kept the `sync` option as this is not too much work to maintain.

**Comment by inkhey** [comment]
@grignards I do agree. So, i suggest:
1. rename theses parameters with a name that can handle all type of workers ?  Maybe "jobs.*" ?
2. use this new config for all new workers including the live_message one.
3. keep sync mode.

**Comment by inkhey** [comment]
After discussion with @lebouquetin , in this issue:
1. Clean and working async mecanism (it's not clear to me what is missing from https://github.com/tracim/tracim/issues/2901)
2. Do or do not rename email.* in jobs.* (depending on how complicated/long is it, it's low priority)
3. wait 3.1 for proper integration of the email https://github.com/tracim/tracim/issues/2901.
